### PR TITLE
JAMES-3597 Exclude deleted messages from JMAP Email/query

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala [hunk 1]
```diff
@@ -206,6 +206,63 @@ trait EmailQueryMethodContract {
     }
   }
 
+  @Disabled("JAMES-3597 Deleted messages are exposed over JMAP")
+  @Test
+  def messagesMarkedAsDeeltedShouldNotBeExposedOverJMAP(server: GuiceJamesServer): Unit = {
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala:211]:
```suggestion
  def messagesMarkedAsDeletedShouldNotBeExposedOverJMAP(server: GuiceJamesServer): Unit = {
```

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala [hunk 2]
```diff
@@ -206,6 +206,63 @@ trait EmailQueryMethodContract {
     }
   }
 
+  @Disabled("JAMES-3597 Deleted messages are exposed over JMAP")
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala:209]: Shouldn't you remove this `Disabled` on your last commit fixing the issue?

Comment (chibenwa) [reply server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala:209]: Crap

**Review by Arsnael — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
(Just rebased)

**Pull request merged**